File: app/src/androidTest/java/com/example/karthickmadasamy/myapplication/view/activity/TestServiceHelper.java
package com.example.karthickmadasamy.myapplication.view.activity;

import android.content.Context;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 Created by Karthick.Madasamy on 12/5/2019.
 Class craated for converting stream to sting & fetch mock json files from assets folder
 */
public class TestServiceHelper {

    public static String convertStreamToString(InputStream is) throws Exception {
        BufferedReader reader = new BufferedReader(new InputStreamReader(is));
        StringBuilder sb = new StringBuilder();
        String line;
        while ((line = reader.readLine()) != null) {
            sb.append(line).append("\n");
        }
        reader.close();
        return sb.toString();
    }

    public static String getStringFromFile(Context context, String filePath) throws Exception {
        final InputStream stream = context.getResources().getAssets().open(filePath);

        String ret = convertStreamToString(stream);
        //Make sure to close all streams.
        stream.close();
        return ret;
    }
}

File: app/src/androidTest/java/com/example/karthickmadasamy/myapplication/view/activity/TestServiceHelper.kt
package com.example.karthickmadasamy.myapplication.view.activity

import android.content.Context
import java.io.BufferedReader
import java.io.InputStream


object TestServiceHelper {

    @Throws(Exception::class)
    fun convertStreamToString(inputStream: InputStream): String {
        val reader = BufferedReader(inputStream.reader())
        val content = StringBuilder()
        try {
            var line = reader.readLine()
            while (line != null) {
                content.append(line)
                line = reader.readLine()
            }
        } finally {
            reader.close()
        }

        return content.toString()
    }

    @Throws(Exception::class)
    fun getStringFromFile(context: Context, filePath: String): String {
        val stream = context.getResources().getAssets().open(filePath)

        val ret = convertStreamToString(stream)
        //Make sure to close all streams.
        stream.close()
        return ret
    }
}